lib/index.ts

```diff
@@ -920,6 +921,10 @@ export function emit(rootDecl: TopLevelDeclaration, { rootFlags = ContextFlags.N
 
             print(p.name);
 
+            if (p.baseType && p.defaultType) {

```

Comment:
Defaults and constraints are not mutually exclusive

Reply:
Oh cool, didn't know this was a thing 😄